vowpalwabbit/learner.h

```diff
@@ -104,23 +121,54 @@ typedef void (*tend_example)(vw& all, void* d, example& ec);
   func_data end_pass_fd;
   func_data end_examples_fd;
   func_data finisher_fd;
+  test_example_data test_example_fd;
 
 public:
+  bool inline multiline_learn() const { return learn_fd.learn_multiline_f != nullptr; }
+  bool inline multiline_predict() const { return learn_fd.predict_multiline_f != nullptr; }
+  bool inline singleline_learn() const { return learn_fd.learn_f != nullptr; }
+  bool inline singleline_predict() const { return learn_fd.predict_f != nullptr; }
+
   prediction_type::prediction_type_t pred_type;
   size_t weights; //this stores the number of "weight vectors" required by the learner.
   size_t increment;
 
   //called once for each example.  Must work under reduction.
-  inline void learn(example& ec, size_t i=0)
-  { ec.ft_offset += (uint32_t)(increment*i);
+  inline void learn(example& ec, size_t i=0) { 
+    ec.ft_offset += (uint32_t)(increment*i);
     learn_fd.learn_f(learn_fd.data, *learn_fd.base, ec);
     ec.ft_offset -= (uint32_t)(increment*i);
   }
-  inline void predict(example& ec, size_t i=0)
-  { ec.ft_offset += (uint32_t)(increment*i);
+  inline void learn(multi_ex& ec_seq, size_t i=0) 
+  {
+    if (ec_seq.size() > 0)
```

Comment:
This seems like it should be an assert rather than a guard---anything producing zeroline multline examples should generate observable errors.